File: low-level-design/design-patterns/golang/creational/factory-pattern/factory/gcp_storage_client.go
package factory

import "fmt"

func NewGCPStorageClient() *GCPStorageClient {	

	return &GCPStorageClient{}
}

type GCPStorageClient struct {
}

func (client *GCPStorageClient) Upload(fileName string, data []byte) error {
	// Implementation for uploading to GCP Storage
	fmt.Println("Uploading to GCP Storage:", fileName, string(data))
	return nil
}

func (client *GCPStorageClient) Download(fileName string) error {
	// Implementation for downloading from GCP Storage
	fmt.Println("Downloading from GCP Storage:", fileName)
	return nil
}

File: low-level-design/design-patterns/python/creational/factory-pattern/gcp_storage_client.py
from storage_client import  StorageClient
class GCPStorageClient(StorageClient):

    def upload(self, content: str, path: str):
        print(f"Uploading {content} to GCP at {path}")

    def download(self, path: str):
        print(f"Downloading from GCP at {path}")
        return f"Content from {path} in GCP"
